src/dmd/mars.d

```diff
@@ -1386,16 +1383,31 @@ bool parseCommandLine(const ref Strings arguments, const size_t argc, ref Param
             params.useDeprecated = Diagnostic.inform;
         else if (arg == "-c")                // https://dlang.org/dmd.html#switch-c
             params.link = false;
-        else if (startsWith(p + 1, "check=")) // https://dlang.org/dmd.html#switch-check
+        else if (startsWith(p + 1, "check") && // https://dlang.org/dmd.html#switch-check
+                *(p + 6) != 'a')            // ignore -checkaction
```

Comment:
Wouldn't it be better to process checkaction before instead ?

Reply:
Done. Thanks!